internal/commands/secrets/list_test.go

```diff
@@ -0,0 +1,181 @@
+package secrets
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/app"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+)
+
+func TestSecretsListSetup(t *testing.T) {
+	t.Run("Should construct a Realm client with the configured base url", func(t *testing.T) {
+		profile := mock.NewProfile(t)
+		profile.SetRealmBaseURL("http://localhost:8080")
+
+		cmd := &CommandList{inputs: listInputs{}}
+		assert.Nil(t, cmd.realmClient)
+
+		assert.Nil(t, cmd.Setup(profile, nil))
+		assert.NotNil(t, cmd.realmClient)
+	})
+}
+
+func TestSecretsListHandler(t *testing.T) {
+	projectID := "projectID"
+	appID := "appID"
+	testApp := realm.App{
+		ID:          appID,
+		GroupID:     projectID,
+		ClientAppID: "eggcorn-abcde",
+		Name:        "eggcorn",
+	}
+
+	t.Run("Should find app secrets", func(t *testing.T) {
+		testSecrets := []realm.Secret{
+			{
+				ID:   "secret1",
```

Comment:
[meganit] just inline these definitions to help reduce the visual load of this test